server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -553,6 +666,119 @@ public Void call() throws Exception {
         processExecutor.execute(command);
     }
 
+    private List<File> buildPipelineExtension(File manifest, Map<String, Object> manifestContext) throws IOException, InterruptedException, ExtenderException {
+
+        if (commonPlatformConfig == null || commonPlatformConfig.protoEngineCxxCmd == null) {
+            LOGGER.info("This SDK version doesn't support compiling plugins!");
+            return new ArrayList<File>();
+        }
+
+        File extDir = manifest.getParentFile();
+        File extBuildDir = createDir(buildDirectory, extDir.getName());
+        extBuildDir.mkdir();
+        File[] srcDirs = { new File(extDir, FOLDER_COMMON_SRC), new File(extDir, FOLDER_PLUGIN_SRC) };
+
+        List<File> protoFiles = listFiles(srcDirs, PROTO_RE);
+
+        List<File> outputFiles = new ArrayList<>();
+
+        // ***************************************************************************
+        // C++
+        {
+            List<File> srcFiles = listFiles(srcDirs, platformConfig.sourceRe);
+
+            if (srcFiles.isEmpty()) {
+                LOGGER.info("No C++ source found for plugin. Skipping {}", extDir);
+            } else {
+                // We leave the C++ protobuf support for a later date
+                List<File> generatedFiles = new ArrayList<>();//generateProtoSrcForPlugin(extDir, manifestContext, protoFiles, "cpp");
+
+                srcFiles.addAll(generatedFiles);
+
+                List<String> objs = new ArrayList<>();
+                List<String> commands = new ArrayList<>();
+
+                // Compile C++ source into object files
+                int i = getAndIncreaseNameCount();
+                for (File src : srcFiles) {
+                    File o = addCompileFileCppShared(i, extDir, src, manifestContext, commands);
+                    objs.add(ExtenderUtil.getRelativePath(jobDirectory, o));
+                    i++;
+                }
+                ProcessExecutor.executeCommands(processExecutor, commands); // in parallel
```

Comment:
Should the name of the function indicate parallelism somehow? So the comments aren't necessary